presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1648,7 +1664,19 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
         List<ColumnHandle> partitionColumns = hiveLayoutHandle.getPartitionColumns();
         List<HivePartition> partitions = hiveLayoutHandle.getPartitions().get();
 
-        TupleDomain<ColumnHandle> predicate = createPredicate(partitionColumns, partitions);
+        TupleDomain<ColumnHandle> predicate;
+        if (partitions.isEmpty()) {
+            predicate = TupleDomain.none();
+        }
+        else {
+            ImmutableMap.Builder<ColumnHandle, Domain> predicateBuilder = ImmutableMap.builder();
+            hiveLayoutHandle.getEffectivePredicate().getDomains()
```

Comment:
To me, here and elsewhere it would be easier to read this if you use instead `ifPresent` and do what you need below, because we're not using the result of `map`.